LoRaEngine/modules/LoRaWan.NetworkServerDiscovery/TagBasedLnsDiscovery.cs

```diff
@@ -24,11 +24,16 @@ public sealed class TagBasedLnsDiscovery : ILnsDiscovery, IDisposable
         private const string NetworkByStationCacheKey = "NetworkByStation";
 
         private static readonly TimeSpan CacheItemExpiration = TimeSpan.FromHours(6);
-        private static readonly IJsonReader<Uri?> HostAddressReader =
+        private static readonly IJsonReader<LnsHostAddressParseResult> HostAddressReader =
             JsonReader.Object(JsonReader.Property("hostAddress",
                                                   from s in JsonReader.String()
-                                                  select string.IsNullOrEmpty(s) ? null : new Uri(s),
-                                                  (true, null)));
+                                                  select Uri.TryCreate(s, default, out var uri) ? uri : null,
+                                                  (true, null)),
+                              JsonReader.Property("deviceId", JsonReader.String()),
+                              (hostAddress, deviceId) => new LnsHostAddressParseResult(hostAddress is { } someHostAddress
+                                                                                                       && someHostAddress.IsAbsoluteUri
+                                                                                                       && someHostAddress.Scheme is "ws" or "wss" ? someHostAddress : null,
```

Comment:
Can't the host address conditions be moved earlier when parsing it?

```suggestion
                                                  select Uri.TryCreate(s, UriKind.Absolute, out var uri)
                                                      && uri.Scheme is "ws" or "wss"
                                                       ? uri : null,
                                                  (true, null)),
                              JsonReader.Property("deviceId", JsonReader.String()),
                              (hostAddress, deviceId) => new LnsHostAddressParseResult(hostAddress, deviceId));
```